KAFKA-12284: wait for mm2 auto-created the topic

showuon opened the pull request:
The reason why the test sometimes failed with: `TopicExistsException: Topic 'primary.test-topic-2' already exists.` is because we tried to create the topic that the MM2 already help us created. That is, 
```java
primary.kafka().createTopic("test-topic-2", NUM_PARTITIONS);
// after the primary cluster created "test-topic-2" topic, the backup cluster will auto-created the topic: "primary.test-topic-2" for us
backup.kafka().createTopic("primary.test-topic-2", 1); // this line will have race condition with the MM2
```

Note: after MM2 replicate the topic, the topic partition number will also get increased to the `NUM_PARTITIONS` in the end. So, we should just let MM2 help us create topic. Also fix 2 issues in the tests:
1. We had resource leak due to no close the adminClient

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

showuon committed:
KAFKA-12284: wait for mm2 auto-created the topic

showuon commented:
@mimaison , could you help review this PR? Thank you.

mimaison commented on connect/mirror/src/test/java/org/apache/kafka/connect/mirror/integration/MirrorConnectorsIntegrationBaseTest.java:
```diff
@@ -455,27 +470,44 @@ private static void waitUntilMirrorMakerIsRunning(EmbeddedConnectCluster connect
+            Set<String> topicsToBeDeleted = adminClient.listTopics().names().get();
+            log.debug("Deleting topics: {} ", topicsToBeDeleted);
+            adminClient.deleteTopics(topicsToBeDeleted).all().get();
+        } catch (final Throwable e) {
```
Why are we catching throwable?

showuon replied:
@mimaison , I just wanted to catch all exceptions here. Do you think it's better we catch specific exceptions here?

mimaison replied:
Catching `Throwable` will also catch errors which I don't think we want to do. But I don't even think we want to catch all exceptions and rethrow them as unchecked. 
The `deleteAllTopics` method is already marked as throwing `Exception`. If anything is thrown, we can let it bubble up and fail the test.

showuon replied:
Make sense to me. Updated. Thanks.

showuon replied:
Also remove the change in PR description

mimaison commented on connect/mirror/src/test/java/org/apache/kafka/connect/mirror/integration/MirrorConnectorsIntegrationBaseTest.java:
```diff
@@ -455,27 +470,44 @@ private static void waitUntilMirrorMakerIsRunning(EmbeddedConnectCluster connect
+     */
+    private static void waitForTopicCreated(EmbeddedKafkaCluster cluster, String topicName) throws InterruptedException {
+        try (final Admin adminClient = cluster.createAdminClient()) {
+            waitForCondition(() -> adminClient.listTopics().names().get().contains(topicName), OFFSET_SYNC_DURATION_MS,
```
Why are we using `OFFSET_SYNC_DURATION_MS`? This does not seem related to offset sync. Maybe we can just use a literal value. WDYT?

showuon replied:
You're right, I created a `TOPIC_SYNC_DURATION_MS` for it.

showuon committed:
KAFKA-12284: don't catch exception in deleteTopics

showuon commented:
@mimaison , thanks for your comments. I've addressed them and updated in this commit: https://github.com/apache/kafka/pull/10185/commits/de95286d9a6463d1fb1ac4b2d5bfd45360aec3a3. Please review again. Thank you.

mimaison reviewed (approved):
Thanks for the PR. LGTM
